src/actions/mod.rs

```diff
@@ -435,6 +435,35 @@ impl ActionHandler {
         }
     }
 
+    pub fn borrow_info<O: Output>(&self, id: usize, params: TextDocumentPositionParams, out: O) {
+        let t = thread::current();
+        let span = self.convert_pos_to_span(&params.text_document, params.position);
+
+        trace!("borrow_info: {:?}", span);
+
+        let analysis = self.analysis.clone();
+        let rustw_handle = thread::spawn(move || {
+            let bi = analysis.borrow_info(&span);
+            t.unpark();
+            bi.map(|b| b.into())
+        });
+
+        thread::park_timeout(Duration::from_millis(::COMPILER_TIMEOUT));
+
+        let result = rustw_handle.join();
+        match result {
+            Ok(Ok(r)) => {
+                out.success(id, ResponseData::BorrowInfo(r));
+            },
+            Ok(Err(e)) => {
+                out.failure(id, &format!("BorrowInfo not available for that location: {:?}", e)[..]);
+            }
+            Err(_) => {
+                out.failure(id, "BorrowInfo failed to complete successfully");
```

Comment:
Other code wasn't logging errors. Should we start to do that? Is there a file we can write to in case writing to stderr interferes with the IDEs?